### ISSUE TITLE
Test set evaluation support?

### ISSUE THREAD

**Issue by aspctu** [allegro/allRank#16]
Is there support to pass in a test set (not to be used as validation set) to assess NCDG?

**Comment by PrzemekPobrotyn** [comment]
Hi, apologies for the late reply!

You can pass the test set instead of the validation set if you want to keep track of NDCG on it during training, we do not recommend doing that though as it may be tempting to overfit to it. 

Alternatively, once you selected your best model based on the validation data and you want to evaluate it on a test set you can use the `rank_slates` function from `allRank/allrank/inference/inference_utils.py` to produce the sorting of your data according to your model and compute NDCG on the result of the sorting.

If that resolves your issues, could you please close it?. Thanks!

**Comment by aspctu** [comment]
Cool, that makes sense. I assumed I could use rank_slates but wanted to see if there was built in support.
